### PR TITLE
Shayan/ FEQ-2388 / replaces localStorage with LocalStorageUtils

### PR DESCRIPTION
**Pull request by shayan-deriv**

### CODE REVIEW

File: src/utils/types.ts [hunk 1]
```diff
@@ -14,3 +14,7 @@ export type TTextColors =
     | 'success'
     | 'warning'
     | 'white';
+
+export type TOrderIdsMap = {
```
Comment (nada-deriv) [inline src/utils/types.ts:18]: can we move this to the other types.ts file ?

File: src/pages/endpoint/screens/Endpoint/Endpoint.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (nada-deriv) [inline src/pages/endpoint/screens/Endpoint/Endpoint.tsx:1]: console ?

File: src/pages/my-ads/screens/MyAds/MyAdsTableRow/MyAdsTableRowView.tsx [hunk 3]
```diff
@@ -155,9 +158,9 @@ const MyAdsTableRowView = ({
 
     useEffect(() => {
         if (isCreateSuccess) {
-            if (shouldNotShowArchiveMessageAgain !== 'true') {
+            if (!shouldNotShowArchiveMessageAgain) {
                 showModal('AdCreateEditSuccessModal');
-            } else if (shouldNotShowArchiveMessageAgain === 'true') {
+            } else if (shouldNotShowArchiveMessageAgain) {
                 invalidate('p2p_advertiser_adverts');
             }
```
Comment (farrah-deriv) [inline src/pages/my-ads/screens/MyAds/MyAdsTableRow/MyAdsTableRowView.tsx:165]:
```suggestion
            if (shouldNotShowArchiveMessageAgain) {
                showModal('AdCreateEditSuccessModal');
            } else { 
                invalidate('p2p_advertiser_adverts');
           }
```

File: src/utils/types.ts (hunk 1 above)
Comment (shayan-deriv) [reply src/utils/types.ts]: done 🙏

File: src/pages/endpoint/screens/Endpoint/Endpoint.tsx (hunk 2 above)
Comment (shayan-deriv) [reply src/pages/endpoint/screens/Endpoint/Endpoint.tsx]: removed

File: src/pages/my-ads/screens/MyAds/MyAdsTableRow/MyAdsTableRowView.tsx (hunk 3 above)
Comment (shayan-deriv) [reply src/pages/my-ads/screens/MyAds/MyAdsTableRow/MyAdsTableRowView.tsx]: done 🙏

File: src/pages/endpoint/screens/Endpoint/__tests__/Endpoint.spec.tsx [hunk 4]
```diff
@@ -3,14 +3,14 @@ import userEvent from '@testing-library/user-event';
 import Endpoint from '../Endpoint';
 
 describe('<Endpoint />', () => {
-    it('should render the endpoint component', () => {
-        render(<Endpoint />);
+    // it('should render the endpoint component', () => {
```
Comment (ameerul-deriv) [inline src/pages/endpoint/screens/Endpoint/__tests__/Endpoint.spec.tsx:6]: why are we commenting out this test case 🤔

Comment (shayan-deriv) [reply src/pages/endpoint/screens/Endpoint/__tests__/Endpoint.spec.tsx]: I was working on fixing the test but it wasn't finished. now this test is passing. there is another test case which I'm working on it right now

File: src/pages/endpoint/screens/Endpoint/__tests__/Endpoint.spec.tsx [hunk 5]
```diff
@@ -19,11 +19,11 @@ describe('<Endpoint />', () => {
...
         const submitButton = screen.getByRole('button', { name: 'Submit' });
 
-        await userEvent.type(serverUrlInput, 'https://example.com');
+        await userEvent.type(serverUrlInput, 'qa10.deriv.dev');
         await userEvent.type(appIdInput, '123');
         await userEvent.click(submitButton);
 
-        expect(localStorage.getItem('config.server_url')).toBe('https://example.com');
-        expect(localStorage.getItem('config.app_id')).toBe('123');
+        expect(JSON.parse(localStorage.getItem('config.server_url') || '')).toBe('qa10.deriv.dev');
```
Comment (niloofar-deriv) [inline src/pages/endpoint/screens/Endpoint/__tests__/Endpoint.spec.tsx:26]: Why you didn't use utils here? 🤔

Comment (shayan-deriv) [reply src/pages/endpoint/screens/Endpoint/__tests__/Endpoint.spec.tsx:26]: this was easier to use

Comment (shayan-deriv) [reply src/pages/endpoint/screens/Endpoint/__tests__/Endpoint.spec.tsx:26]: with utils we have to mock it

Comment (niloofar-deriv) [reply src/pages/endpoint/screens/Endpoint/__tests__/Endpoint.spec.tsx:26]: Ok then got it 👍

**Review by ameerul-deriv — APPROVED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Pull request merged**